cmplibrary/src/test/java/com/sourcepoint/gdpr_cmplibrary/SourcePointClientTest.java

```diff
@@ -1,262 +1,250 @@
-//            assertNotNull(e);
-//            verify(callback).onFailure(e);
-//        }
-//
-//        ArgumentCaptor<ConsentLibException> captor = ArgumentCaptor.forClass(ConsentLibException.class);
-//        verify(callback, times(1)).onFailure(captor.capture());
-//    }
-//
-//    @Test
-//    public void sendConsent_onFailure() throws Exception {
-//
-//        GDPRConsentLib.OnLoadComplete callback = mock(GDPRConsentLib.OnLoadComplete.class);
-//
-//        //sourcePointClientMock.baseMsgUrl = mockWebServer.url("/").toString();
-//
-//        String response = "{\"messageUrl\":\"http://google.com\"}";
-//        JSONObject jsonObject = new JSONObject(response);
-//
-//        sourcePointClientMock.setHttpDummy( new SyncHttpClient());
-//        mockWebServer.enqueue(new MockResponse().setResponseCode(500).setBody("Server Error"));
-//        sourcePointClientMock.sendConsent(jsonObject,callback); // calling the method under test
-//
-//        ArgumentCaptor<ConsentLibException> captor = ArgumentCaptor.forClass(ConsentLibException.class);
-//        verify(callback, times(1)).onFailure(captor.capture());
-//    }
-//}
\ No newline at end of file
+package com.sourcepoint.gdpr_cmplibrary;
+
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+import android.os.Build;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+
+import okhttp3.OkHttpClient;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.timeout;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+
+@RunWith(RobolectricTestRunner.class)
+@Config(manifest = Config.NONE, sdk = Build.VERSION_CODES.O_MR1)
+public class SourcePointClientTest {
+
+    private OkHttpClient http = new OkHttpClient();
+    private SourcePointClient sourcePointClientMock;
+    private GDPRConsentLib.OnLoadComplete onLoadComplete;
+
+    private MockWebServer webServerMock;
+
+    private String consentUUID = "consentUUID";
+    private String euConsent = "euConsent";
+    private String meta = "meta";
+    private boolean isNative = false;
+    private NetworkInfo networkInfo;
+    private ConnectivityManager connectivityManager;
+
+
+    @Before
+    public void setUp() throws IOException {
+        onLoadComplete = mock(GDPRConsentLib.OnLoadComplete.class);
```

Comment:
you can use annotations for the mocks :)

Reply:
By using mock annotation we get NPE.